addons/binding/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/config/OpenSprinklerConfig.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.opensprinkler.config;
+
+import static org.openhab.binding.opensprinkler.OpenSprinklerBindingConstants.DEFAULT_REFRESH_RATE;
+import static org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApiConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The {@link OpenSprinklerConfig} class defines the configuration options
+ * for the OpenSprinkler Thing.
+ *
+ * @author Chris Graham - Initial contribution
+ */
+public class OpenSprinklerConfig {
+    /**
+     * Hostname of the OpenSprinkler API.
+     */
+    public String hostname = null;
+
+    /**
+     * The port the OpenSprinkler API is listening on.
+     */
+    public int port = DEFAULT_API_PORT;
+
+    /**
+     * The password to connect to the OpenSprinkler API.
+     */
+    public String password = DEFAULT_ADMIN_PASSWORD;
+
+    /**
+     * Number of seconds in between refreshes from the OpenSprinkler device.
+     */
+    public int refresh = DEFAULT_REFRESH_RATE;
+
+    /**
+     * Returns the properties of this class as a string object tuple map.
+     *
+     * @return {@link HashMap} of String Object tuples of the properties.
+     */
+    public Map<String, Object> getPropertyMap() {
```

Comment:
I think this method can be removed - it is only used at a single place and there it does not even make much sense...